src/access/AccessRegistry.sol

```diff
@@ -0,0 +1,160 @@
+    modifier onlyNonAccessPointOwner(address user) {
+        IAccessPoint accessPoint = _accessPoints[user];
+        if (address(accessPoint) != address(0)) {
+            revert UserHasAccessPoint(user, accessPoint);
+        }
+        _;
+    }
+
+    /* ============ Constructor & Upgrades ============ */
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * @dev Upgrade calling `upgradeTo()`
+     */
+    function initialize(IAccessPoint impl) external virtual initializer {
+        __Ownable_init();
+        __UUPSUpgradeable_init();
+        beacon = new UpgradeableBeacon(address(impl));
+        factoryVersion = 1;
+    }
+
+    /**
+     * @dev Upgrade the wallet implementations using the beacon
+     * @param newImpl The new implementation
+     */
+    function upgradeAll(IAccessPoint newImpl) external override onlyOwner {
+        require(address(newImpl) != address(0) && address(newImpl) != beacon.implementation(), "invalid address");
+        factoryVersion++;
+        emit AccessPointFactoryUpgraded(beacon.implementation(), address(newImpl));
+        beacon.upgradeTo(address(newImpl));
+    }
+
+    /**
+     * @dev Authorize the upgrade. Only by an owner.
+     * @param newImplementation address of the new implementation
+     */
+    // This function is called by the proxy contract when the factory is upgraded
+    function _authorizeUpgrade(address newImplementation) internal view override onlyOwner {
+        (newImplementation);
+    }
+
+    /* ============ View Functions ============ */
+
+    /// @inheritdoc IAccessRegistry
+    function isWorkflowAllowed(address workflow) external view returns (bool) {
+        return _workflows[workflow];
+    }
+
+    /// @inheritdoc IAccessRegistry
+    function getAccessPoint(address user) external view returns (IAccessPoint accessPoint) {
+        accessPoint = _accessPoints[user];
+    }
+
+    /// @inheritdoc IAccessRegistry
+    function getAddress(address owner) public view override returns (address) {
+        return Create2.computeAddress(
+            bytes32(abi.encodePacked(owner)),
+            keccak256(
+                abi.encodePacked(
+                    type(SafeBeaconProxy).creationCode,
+                    abi.encode(address(beacon), abi.encodeCall(IAccessPoint.initialize, (owner)))
+                )
+            )
+        );
+    }
+
+    /// @dev TODO:Should be removed. Added because Kinto EntryPoint needs this function.
+    function getWalletTimestamp(address) external pure returns (uint256) {
+        return 1;
+    }
+
+    /* ============ State Change ============ */
+
+    /// @inheritdoc IAccessRegistry
+    function disallowWorkflow(address workflow) external onlyOwner {
+        _workflows[workflow] = false;
+        emit WorkflowStatusChanged(workflow, false);
+    }
+
+    /// @inheritdoc IAccessRegistry
+    function allowWorkflow(address workflow) external onlyOwner {
+        _workflows[workflow] = true;
```

Comment:
same, require workflow is disabled

Reply:
we could potetially also merge `allowWorkflow` with `disallowWorkflow` and have only one `function allowWorkflow(address worfklow, bool allowed)`

Reply:
>we could potetially also merge allowWorkflow with disallowWorkflow and have only one function allowWorkflow(address worfklow, bool allowed)

In my experience users are always confused by `bool allowed` and it is better to have two external functions and one private. I'll make a change.

Reply:
I kept them separate for clarity but not strong opinion. Feel to modify to your liking.

Reply:
sgtm, not strong opinion neither